lib/tesla/middleware/retry.ex

```diff
@@ -29,22 +38,21 @@ defmodule Tesla.Middleware.Retry do
     opts = opts || []
     delay = Keyword.get(opts, :delay, @defaults[:delay])
     max_retries = Keyword.get(opts, :max_retries, @defaults[:max_retries])
+    should_retry = Keyword.get(opts, :should_retry, &match?({:error, _}, &1))
```

Comment:
I have tried to move it to default before. But attribute does not support for function capturing...
Here are the print stack
```
== Compilation error in file lib/tesla/middleware/retry.ex ==
** (ArgumentError) cannot inject attribute @defaults into function/macro because cannot escape #Function<0.103079773 in file:lib/tesla/middleware/retry.ex>. The supported values are: lists, tuples, maps, atoms, numbers, bitstrings, PIDs and remote functions in the format &Mod.fun/arity
    (elixir) lib/kernel.ex:2782: Kernel.do_at/5
    (elixir) expanding macro: Kernel.@/1
    lib/tesla/middleware/retry.ex:40: Tesla.Middleware.Retry.call/3
    (elixir) lib/kernel/parallel_compiler.ex:206: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/6
```